### PR TITLE
Decompress packages and create idx files

### PR DESCRIPTION
**Pull request by rtaket**

### CODE REVIEW

File: src/Ng/ElfieFromCatalogCollector.cs [hunk 1]
```diff
@@ -221,5 +230,89 @@ public ElfieFromCatalogCollector(Uri index, Storage storage, int maxThreads, Fun
...
+        /// </summary>
+        async Task<Uri> DecompressAndIndexPackageAsync(Uri packageResourceUri, CatalogItem catalogItem, CancellationToken cancellationToken)
+        {
+            Trace.TraceInformation("#StartActivity DecompressAndIndexPackageAsync " + catalogItem.PackageId + " " + catalogItem.PackageVersion);
+
+            Uri idxResourceUri = null;
+
+            // This is the pointer to the package file in storage
+            Trace.TraceInformation("Loading package from storage.");
+            StorageContent packageStorage = this._storage.Load(packageResourceUri, new CancellationToken()).Result;
```
Comment (jinujoseph) [inline src/Ng/ElfieFromCatalogCollector.cs:245]: so we store the packages in blob storage ?

File: src/Ng/Models/RegistrationItem.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+using System.Web;
+using Newtonsoft.Json;
+using Catalog = NuGet.Services.Metadata.Catalog;
+
+namespace Ng.Models
+{
+    /// <summary>
+    /// Model for a NuGet registration item, https://api.nuget.org/v3/registration1/newtonsoft.json/8.0.2.json.
+    /// </summary>
+    public class RegistrationItem
```
Comment (jinujoseph) [inline src/Ng/Models/RegistrationItem.cs]: was not this file part of your last PR ?

**Comment by jinujoseph** [comment]
Over all LGTM, lot of code change so might have missed some thing. i don't think we need to check in all the Elfie binaries , but we can clean that later. 
or it should all go away with the nuget package for Elfie.

File: src/Ng/ElfieFromCatalogCollector.cs (hunk 1 above)
Comment (rtaket) [inline src/Ng/ElfieFromCatalogCollector.cs:245]: We do store the packages in blob storage. I was intending to store the package, idx, ardb(txt) and patches in blob storage.
It's not a requirement that we store the packages though. I was just keeping them for debugging purposes (if the idx creation fails, we'd have the exact package we were trying to index for investigation.) We can not store them if we end up never using them.

File: src/Ng/Models/RegistrationItem.cs (hunk 2 above)
Comment (rtaket) [inline src/Ng/Models/RegistrationItem.cs]: Umm, yes, this file was part of the last PR. I was supposed to delete it. I'm not sure if I merged incorrectly, but I don't know how this file made it back in. I will attempt to remove it again. Sorry about that.

**Pull request merged**